app/src/main/java/net/squanchy/proximity/near/NearITProximityService.java

```diff
@@ -0,0 +1,49 @@
+package net.squanchy.proximity.near;
+
+import android.os.Parcelable;
+
+import net.squanchy.proximity.ProximityEvent;
+import net.squanchy.proximity.ProximityProvider;
+import net.squanchy.proximity.ProximityService;
+
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import io.reactivex.schedulers.Schedulers;
+import it.near.sdk.Geopolis.Beacons.Ranging.ProximityListener;
+import it.near.sdk.Reactions.CustomJSON.CustomJSON;
+import it.near.sdk.Reactions.CustomJSON.CustomJSONReaction;
+import it.near.sdk.Recipes.Models.Recipe;
+
+public class NearITProximityService implements ProximityService {
+
+    private static final String PROXIMITY_ACTION = "action";
+    private static final String SUBJECT_ID = "subject";
+    private final NearITProximityProvider nearITProximityProvider;
+
+    public NearITProximityService(ProximityProvider proximityProvider) {
+        this.nearITProximityProvider = (NearITProximityProvider) proximityProvider;
```

Comment:
If we then rely to the particular implementation here we lose all the benefit of having an interface, please pull the methods you require in the interface so that you don't depend on a specific implementation (also because this is a runtime crash if we decide to change `ProximityProvider` implementation).

Remember also to rename the field once you change the type required.